components/playlist/playlist_service.h

```diff
@@ -0,0 +1,161 @@
+                  PlaylistYoutubeDownComponentManager* manager);
+  ~PlaylistService() override;
+  PlaylistService(const PlaylistService&) = delete;
+  PlaylistService& operator=(const PlaylistService&) = delete;
+
+  base::Value GetAllPlaylistItems();
+  base::Value GetPlaylistItem(const std::string& id);
+  void RecoverPlaylistItem(const std::string& id);
+  void DeletePlaylistItem(const std::string& id);
+  void DeleteAllPlaylistItems();
+  void RequestDownload(const std::string& url);
+
+  void AddObserver(PlaylistServiceObserver* observer);
+  void RemoveObserver(PlaylistServiceObserver* observer);
+
+  bool GetThumbnailPath(const std::string& id, base::FilePath* thumbnail_path);
+
+  base::FilePath GetPlaylistItemDirPath(const std::string& id) const;
+
+ private:
+  FRIEND_TEST_ALL_PREFIXES(PlaylistBrowserTest, CreatePlaylist);
+  FRIEND_TEST_ALL_PREFIXES(PlaylistBrowserTest,
+                           CreatePlaylistWithSeparateAudio);
+  FRIEND_TEST_ALL_PREFIXES(PlaylistBrowserTest, ThumbnailFailed);
+  FRIEND_TEST_ALL_PREFIXES(PlaylistBrowserTest, MediaDownloadFailed);
+  FRIEND_TEST_ALL_PREFIXES(PlaylistBrowserTest, ApiFunctions);
+
+  // KeyedService overrides:
+  void Shutdown() override;
+
+  // PlaylistMediaFileDownloadManager::Delegate overrides:
+  // Called when all audio/video media file are downloaded.
+  void OnMediaFileReady(const std::string& id,
+                        const std::string& audio_file_path,
+                        const std::string& video_file_path) override;
+  void OnMediaFileGenerationFailed(const std::string& id) override;
+  bool IsValidPlaylistItem(const std::string& id) override;
+
+  // PlaylistThumbnailDownloader::Delegate overrides:
+  // Called when thumbnail image file is downloaded.
+  void OnThumbnailDownloaded(const std::string& id,
+                             const base::FilePath& path) override;
+
+  // PlaylistDownloadRequestManager::Delegate overrides:
+  // Called when meta data is ready. |params| have playlist item's audio/video
+  // media files url, thumbnail and title.
+  void OnPlaylistCreationParamsReady(
+      const CreatePlaylistParams& params) override;
+
+  void OnPlaylistItemDirCreated(const std::string& id, bool directory_ready);
+
+  void CreatePlaylistItem(const CreatePlaylistParams& params);
+  void DownloadThumbnail(const std::string& id);
+  void GenerateMediafileForPlaylistItem(const std::string& id);
+
+  base::SequencedTaskRunner* task_runner();
+
+  // Delete orphaned playlist item directories that are not included in db.
+  void CleanUp();
+  void OnGetOrphanedPaths(const std::vector<base::FilePath> paths);
+
+  void NotifyPlaylistChanged(const PlaylistChangeParams& params);
+
+  void UpdatePlaylistValue(const std::string& id, base::Value value);
+  void RemovePlaylist(const std::string& id);
+
+  // index.html is only used for demo (brave://playlist)
+  void GenerateIndexHTMLFile(const base::FilePath& playlist_path);
```

Comment:
I think we don't need to generate this anymore.
Each playlist item has its own html file for playing. It was only for demo purpose.
In production, new webui will have whole ui for managing/playing playlist items.

Reply:
Yep . Done.